CRM/Case/XMLProcessor/Process.php

```diff
@@ -634,6 +634,10 @@ public function getListeners($caseType) {
    * @return int
    */
   public function getRedactActivityEmail() {
+    $setting = Civi::settings()->get('civicaseRedactActivityEmail');
```

Comment:
Well, sometimes a little duplication is better. It seems like a placing a bet for where you think it'll go.

In this case... the functions are not quite identical. One guards explicitly against the condition where `$xml` is unavailable, and the others don't. This feels like a mistake -- the guard seems sensible for all of them. Which is the kind of problem you'd expect from too much duplication... so I agree on consolidating...